src/os/bluestore/BlueStore.cc

```diff
@@ -3563,12 +3563,11 @@ int BlueStore::_set_cache_sizes()
   double cache_kv_max_ratio = 0;
 
   // if cache_kv_max is negative, disable it
-  if (cache_kv_max > 0) {
-    if (cache_size > 0)
-      cache_kv_max_ratio = (double) cache_kv_max / (double) cache_size;
+  if (cache_size > 0 && cache_kv_max > 0) {
```

Comment:
I think zero is a valid value for cache_kv_max hence it should be >=0.

Another question - do you intend original cache_kv_ratio to be applied when cache_kv_max is negative? Or negative value just means cache_kv_max = 0 (and hence cache_kv_ratio should be reduced to 0 as well)? Current implementation tend to the first case...